pymongo/pool.py

```diff
@@ -550,7 +550,7 @@ def unpin(self):
             self.close_socket(ConnectionClosedReason.STALE)
 
     def hello_cmd(self):
-        if self.opts.server_api or self.hello_ok:
+        if self.opts.server_api or self.hello_ok or self.opts.load_balanced:
```

Comment:
We still need to change it to use OP_MSG for the initial handshake.

Reply:
I believe it does use it for the initial handshake. It is called in the same file
```
def _hello(self, cluster_time, topology_version, heartbeat_frequency):
        cmd = self.hello_cmd()
```
And then it modifies command but when it performs the initial handshake it uses that command which is instantiated from that function:
```
doc = self.command('admin', cmd, publish_events=False,
                           exhaust_allowed=awaitable)
        # PYTHON-2712 will remove this topologyVersion fallback logic.
        if self.opts.load_balanced and _MOCK_SERVICE_ID:
            process_id = doc.get('topologyVersion', {}).get('processId')
            doc.setdefault('serviceId', process_id)
        if not self.opts.load_balanced:
            doc.pop('serviceId', None)
        hello = Hello(doc, awaitable=awaitable)
        self.is_writable = hello.is_writable
        self.max_wire_version = hello.max_wire_version
        self.max_bson_size = hello.max_bson_size
        self.max_message_size = hello.max_message_size
        self.max_write_batch_size = hello.max_write_batch_size
```
I also took a look and it appears that the `hello` function is being called from the connect function:
```
def connect(self):
        sock_info = SocketInfo(sock, self, self.address, conn_id)
        try:
            if self.handshake:
                sock_info.hello()
```

Reply:
The hello command != OP_MSG. This change is sending the initial hello command with OP_QUERY. Take a look at where we decide how to use op_msg vs op_query.